### ISSUE TITLE
SIM118 false positive when using awsglue.dynamicframe.DynamicFrameCollection

### ISSUE THREAD

**Issue by robmcd** [MartinThoma/flake8-simplify#175]
With code like:

```
for dyf_name in dfc.keys():
    print(dyf_name)
```

Where dfc is an instance of [DynamicFrameCollection](https://docs.aws.amazon.com/glue/latest/dg/aws-glue-api-crawler-pyspark-extensions-dynamic-frame-collection.html)

We see:
> SIM118 [*] Use `dyf_name in dfc` instead of `dyf_name in dfc.keys()`

However, the suggested change will produce a `KeyError`.

Is there some way to fix the recommendation to avoid these false positives (apart from ignoring the rule/line)?
